common/configuration/aws_secrets.go

```diff
@@ -0,0 +1,216 @@
+package configuration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"sync"
+
+	"github.com/TBD54566975/ftl/internal/slices"
+
+	"github.com/alecthomas/types/optional"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager/types"
+	"github.com/aws/smithy-go"
+)
+
+const schemeKey = "asm"
```

Comment:
Because the package contains a bunch of different providers/resolvers, this should be prefixed with `asmSchemeKey`... although I'd probably just inline it as it's only used in two locations.